src/reducers/walletReducer.js

```diff
@@ -34,24 +35,34 @@ export default function (state = initState, action) {
   switch (action.type) {
     // metamask
     case 'CHECK_METAMASK_CONNECTION_FULFILLED':
-      return update(state, { wallet: { metamask: { $set: action.payload } } })
+      return update(state, { wallet: { metamask: { $merge: action.payload } } })
```

Comment:
metmask payload has to overwrite wallet data, no touch

Reply:
This will overwrite existing data, but keeps other data that's not in the payload.